### PR TITLE
feature/disable-users-account

### PR DESCRIPTION
**Pull request by darrylpargeter**
adds the feature requested in #56 

I have added the process for disabling a user if the script can't find the code in the user GitHub or if the file is over the size limit.

the fields currently get updated but it still performs a get all query but I have left a query (line 21) that will only get non-disabled accounts when you want to go live

### CODE REVIEW

**Comment by Asuza** [comment]
Merging in the master branch and pushing should resolve the CI issue above. I'll look at your PR in detail as soon as I have time.

**Comment by darrylpargeter** [comment]
That has been merged sorry about that.

**Comment by Asuza** [comment]
I broke the build in master before your PR. It's my fault entirely. Thanks for updating it.

File: gamerunner/user_code/get-user-code-from-github.js [hunk 1]
```diff
@@ -103,17 +108,82 @@ function retrieveCode (users, category) {
...
+    const q = `UPDATE player SET disabled = $1, disabled_count = $2, disabled_message = $3, ${date} WHERE github_id = $4`;
+    const disabled = user.disabled_count >= 2 ? true : false;
+    const disabledCount = ++user.disabled_count;
+    const values = success ? [false, 0, null, user.githud_id] : [disabled, disabledCount, error, user.github_id];
+    db.update(q, values, (err, update) => {
+        count--;
+        if (err) {
+            throw err;
+        }
+        if (count === 0) {
```
Comment (Asuza) [inline gamerunner/user_code/get-user-code-from-github.js]: I think this needs to be before the error gets thrown. If an error occurred during the last update, I'm not sure the DB connection would close as-is.

**Comment by Asuza** [comment]
Thanks, @darrylpargeter. As soon as I have time, I'll test out your changes.

**Comment by Asuza** [comment]
It looks like something's wrong. I checked out the PR, destroyed my existing database and re-initialized it as so:
```
vagrant up
vagrant ssh
cd /vagrant
node database/destroy.js --proceed
node database/initialize.js
psql jsfightclub
```
Within psql, I added two users. One I know does not have an AI and one I know that does.
```
insert into player (github_login) values ('hieumediavn');
insert into player (github_login) values ('Asuza');
\q
```
Then, when trying to fetch code, there's a script error:
```
vagrant@vagrant-ubuntu-trusty-64 /vagrant> node gamerunner/user_code/get-user-code-from-github.js 
About to retrieve user code from Github...
Saving code for hieumediavn / helpers
Saving code for hieumediavn / hero
TypeError: Cannot read property 'last_update_date' of undefined
    at getUser.then (/vagrant/gamerunner/user_code/get-user-code-from-github.js:135:40)
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)
TypeError: Cannot read property 'last_update_date' of undefined
    at getUser.then (/vagrant/gamerunner/user_code/get-user-code-from-github.js:135:40)
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)
Saving code for Asuza / hero
Saving code for Asuza / helpers
```

**Comment by darrylpargeter** [comment]
I shall have a look into this tomorrow.

**Comment by darrylpargeter** [comment]
After having a quick look, the reason for the above issue is that I've set the code to use `github_id` and not `github_login`, as I assumed that `github_id` would always have a value. I can change to `github_login` if you prefer to use `github_login`?

also, I noticed an issue with the `db.close` being called early, but I have managed to fix that.
so I shall hold off updating the PR until I know which you prefer.

**Comment by Asuza** [comment]
`github_login` is preferred. I was only storing the id in case someone changed their GitHub login, I would be able to identify the change. I'm not sure the id is used anywhere. Thanks!

**Comment by Asuza** [comment]
Checked it out again and retried it. The end result is that the script says it's saving hero/helper code, but it never actually happens.

Here's the output:
```
vagrant@vagrant-ubuntu-trusty-64 /vagrant> node gamerunner/user_code/get-user-code-from-github.js 
About to retrieve user code from Github...
Saving code for hieumediavn / hero
Saving code for hieumediavn / helpers
Saving code for Asuza / hero
Saving code for Asuza / helpers
```
But if you look in the directory where hero code goes, only code for Asuza exists.

I don't know if you've used vagrant before, but it makes it pretty easy to get up and running with the database so you can test these things. I've included a lot of information in the [README](https://github.com/JSJitsu/ai-battle-website/blob/master/README.md) to get going. If you don't want to do that, writing unit tests for some of these changes might make it easier to catch any problems without setting up much of an environment. #66 has set up some of the framework to do unit tests.

**Comment by darrylpargeter** [comment]
Just to double check, you mentioned that out of the two users only one was expected to work.
Which when I was testing was Asuza and that seems to match up with your results as well. So is it now the case that hieumediavn should be returning files as well or is it just a poorly placed console?

**Comment by Asuza** [comment]
hieumediavn doesn't have a hero-starter repo, so it should be showing an error as well as increasing the failure count that you put in place.

I'm fairly certain that the count never increased in the database, but I don't have access to that machine to check at the moment.

**Pull request closed without merging**